dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -100,7 +118,50 @@ def _prepare_report(report, output_format=None, omit_keys=None):
         if isinstance(value, set):
             value = sorted(list(value))
 
-        if isinstance(value, dict):
+        # For data_stats (in structured case), need to recurse through a list
+        # As well as account for omit_keys containing indices instead of keys
+        if key == "data_stats" and isinstance(value, list):
+            fmt_report["data_stats"] = []
+            # split off any remaining keys for the recursion
+            # i.e. [test0, test1.test2] -> omit_keys => [test1.test2]
+            # This will filter out all the data_stats omit keys
+            data_stat_omit_keys = []
+            for omit_key in omit_keys:
+                omit_key_split = omit_key.split('.', 1)
+
+                # Must have more keys left for recursion
+                if len(omit_key_split) > 1:
+                    next_key_layer = omit_key_split[-1]
+                    prior_key_layer = omit_key_split[0]
+                    if len(next_key_layer) > 0:
+                        if prior_key_layer == '*' or prior_key_layer == key:
+                            data_stat_omit_keys.append(next_key_layer)
+
+            # Recurse through indices in data_stats and call report_helper
+            for i in range(len(value)):
+                i_index_omit_keys = []
+                # Filter off data_stats omit keys depending on index
+                for nl_key in data_stat_omit_keys:
+                    key_split = nl_key.split(".", 1)
+
+                    # Must have more keys left for recursion
+                    if len(key_split) > 1:
+                        next_key_layer = key_split[-1]
+                        index = key_split[0]
+                        if len(next_key_layer) > 0:
+                            if (index.isdigit() and int(index) == i) \
+                                    or index == "*":
+                                i_index_omit_keys.append(next_key_layer)
+
+                # Recursively prepare each column in data_stats list
+                fmt_report["data_stats"].append(
+                    _prepare_report(value[i], output_format, i_index_omit_keys))
```

Comment:
if the above string for omits is unchanged, at this point we could have something like:
```python
"data_stats.TEST1.statistics.times",
"data_stats.TEST2.statistics.avg_predictions",
"data_stats.*.statistics.data_label_representation",
```

In this case would something like this work?
```python
# if * then nothing else should exist
if '*' in omit_keys:
    continue

# recurse through indices calling prepare
for col_ind in range(value):
    # column is being omitted
    if str(report[key][col_ind].get("column_name")) in omit_keys:
        continue
    
    # update omit keys
    next_layer_omit_keys = []
    for omit_key in omit_keys:
       omit_key_split = omit_key.split('.', 1)
       
       # this omit occurs only at this level, don't include
       if len(omit_key_split) == 1:
           continue
           
        # this omit applies to everything below where more matching can occur
        elif omit_key[:2] == '*.':
            next_layer_omit_keys.append(omit_key[2:]
            
        # this omit validates that the column name matches to get included
        elif omit_key_split[0] == str(report[key][col_ind].get("column_name")):
            next_layer_omit_keys.append(omit_key_split[1]
    
    report[key][col_ind]._prepare_report(report[key][col_ind], output_format, next_layer_omit_keys))
```

Reply:
Just trying to avoid the omit key change additions and reverts

Reply:
Also if we didn't want to do a columns to keep or a list at the end, can reverse the for loop and pop it from the list
e.g. `report[key].pop()` followed by a `continue`